### PR TITLE
nodejs,nodejs-lts: npm: set `foreground-scripts` to true on postinstall

### LINKED ISSUE #3188: create-react-app error: cb() never called!
**Problem description**
Using `create-react-app` after having installed with `npm i -g create-react-app` results in a fatal error.
I saw several people on the Internet recommending to erase the cache, but this workaround did not work.

**Steps to reproduce**
1. `pkg i nodejs`
2. `npm i -g create-react-app`
3. `create-react-app app`

**Expected behavior**
I would have expected a bootstrapped React app

**Additional information**
```bash
$ termux-info
Updatable packages:
All packages up to date
System information:
Linux localhost 4.4.23+ #1 SMP PREEMPT Tue Dec 4 09:52:16 CST 2018 aarch64 Android
Termux-packages arch:
aarch64
Android version:
8.0.0
Device manufacturer:
HUAWEI
Device model:
SHT-W09
```

photo_2019-01-06_14-27-38

### LINKED ISSUE #8387: [Bug]: npm install Exit handler never called!
### Problem description

I use `npm install` and I get this:

```
npm ERR! Exit handler never called!

npm ERR! This is an error with npm itself. Please report this error at:
npm ERR!     <https://github.com/npm/cli/issues>

npm ERR! A complete log of this run can be found in:
npm ERR!     /data/data/com.termux/files/home/.npm/_logs/2021-12-29T01_28_01_927Z-debug.log
```

when I ignore it and run `npm run serve`, I get this: 

```
node:internal/modules/cjs/loader:936
  throw err;
  ^

Error: Cannot find module '/data/data/com.termux/files/home/Adachi-BOT/...'
    at Function.Module._resolveFilename (node:internal/modules/cjs/loader:933:15)                                   at Function.Module._load (node:internal/modules/cjs/loader:778:27)
    at Function.executeUserEntryPoint [as runMain] (node:internal/modules/run_main:81:12)
    at node:internal/main/run_main_module:17:47 {         code: 'MODULE_NOT_FOUND',
  requireStack: []
}
```

### What steps will reproduce the bug?

- run `npm i`
- run `npm run serve`

### What is the expected behavior?

✔ All packages installed (590 packages installed from npm registry, used 23s(network 8s), speed 3.32MB/s, json 516(3.12MB), tarball 23.35MB)

### System information

termux-info:

```
Application version:
0.117
Packages CPU architecture:
aarch64                                                 Subscribed repositories:
# sources.list
deb https://termux.mentality.rip/termux-main/ stable main
# game-repo (sources.list.d/game.list)                  deb https://packages.termux.org/apt/termux-games games stable
# science-repo (sources.list.d/science.list)            deb https://packages.termux.org/apt/termux-science science stable
Updatable packages:                                     All packages up to date
Android version:
11
Kernel build information:
Linux localhost 5.4.86-qgki-g35572e50a678 #1 SMP PREEMPT Wed Oct 27 03:18:25 CST 2021 aarch64 Android
Device manufacturer:
Xiaomi
Device model:
M2102K1AC
```

I use `nodejs-lts`
- node: `v16.13.1`
- npm: `8.1.2`

I try use `nodejs`, but I still get it.
- node: `v17.3.0`
- npm: `8.3.0`

### PR DESCRIPTION
**Pull request by thunder-coding**
Temporary fix for #3188 and #8387

Closes #3188 
Closes #8387

As noted in https://github.com/npm/statusboard/issues/395#issuecomment-1003256492, this seems to be a problem with how npm handles postinstall, preinstall and other package hooks. Adding `--foreground-scripts` is a temporary hack until itis fixed.

### CODE REVIEW

**Comment by thunder-coding** [comment]
@truboxl, I will appreciate a review of yours for the npm exit handler workaround for emscripten

File: packages/emscripten/build.sh [hunk 1]
```diff
@@ -235,13 +235,24 @@ termux_step_make_install() {
...
 	if [ -n "\$(command -v npm)" ]; then
-	echo 'Running "npm ci --no-optional --production" in $TERMUX_PREFIX/opt/emscripten ...'
 	cd "$TERMUX_PREFIX/opt/emscripten"
+	NPM_VERSION=\$(npm --version)
+	NPM_MAJOR_VERSION=\${NPM_VERSION:0:1}
+	if [ 6 = \$NPM_MAJOR_VERSION ]; then
+	echo 'Running "npm ci --no-optional --production" in $TERMUX_PREFIX/opt/emscripten ...'
 	npm ci --no-optional --production
 	else
+	echo 'Running "npm install" in $TERMUX_PREFIX/opt/emscripten ...'
```
Comment (truboxl) [inline packages/emscripten/build.sh:251]:
```suggestion
	echo 'Running "npm install --no-optional --production" in $TERMUX_PREFIX/opt/emscripten ...'
```

File: packages/emscripten/build.sh [hunk 2]
```diff
@@ -235,13 +235,24 @@ termux_step_make_install() {
...
+	rm package-lock.json
+	npm install
```
Comment (truboxl) [inline packages/emscripten/build.sh:253]:
```suggestion
	npm install --no-optional --production
```

File: packages/emscripten/build.sh [hunk 3]
```diff
@@ -235,13 +235,24 @@ termux_step_make_install() {
 }
 
 termux_step_create_debscripts() {
+	# emscripten's package-lock.json is generated with nodejs v12.13.0
+	# which comes with npm v6 which used lockfile version 1
+	# which isn't compatible with lockfile version 2 used in npm v7 and v8
 	cat <<- EOF > postinst
-	#!$TERMUX_PREFIX/bin/sh
+	#!$TERMUX_PREFIX/bin/bash
```
Comment (truboxl) [inline packages/emscripten/build.sh:242]: I assume NPM_MAJOR_VERSION need this? Ok I guess

**Review by truboxl — COMMENTED** [review]
Suggest these changes. Thanks @thunder-coding

**Comment by truboxl** [comment]
Unfortunarely building emscripten will add an extra 2 hours. Suggest separate into 2 PR if build timeout

File: packages/emscripten/build.sh (hunk 3 above)
Comment (thunder-coding) [reply packages/emscripten/build.sh:242]: Yeah

**Comment by thunder-coding** [comment]
> Unfortunarely building emscripten will add an extra 2 hours. Suggest separate into 2 PR if build timeout

Don't worry, I have confirmed that nodejs and nodejs-lts are building and also emscripten according to previous action runs. While merging I will add `%ci:no-build` to commit message and trigger builds for nodejs, nodejs-lts and emscripten manually so that these changes get reflected as soon as possible

**Comment by thunder-coding** [comment]
I will merge this tommorow after 7 AM IST (i.e. 1:30 AM GMT). Till then code reviews are most welcome

**Pull request merged**